libs/blocks/merch-cards/merch-cards.js

```diff
@@ -65,25 +66,24 @@ export function parsePreferences(elements) {
   });
 }
 
-async function initMerchCards(config, type, filtered, el, preferences) {
-  let cardsData;
-  let err;
+const localizedPath = (path, config) => `${config?.locale?.prefix ?? ''}${path}`;
 
+const fetchOverrideCard = (path, config) => new Promise((resolve, reject) => {
   try {
-    const res = await fetch(`${config?.locale?.prefix ?? ''}${config.queryIndexCardPath}.json?sheet=${type}`);
-    if (res.ok) {
-      cardsData = await res.json();
-    } else {
-      err = res.statusText || res.status;
-    }
+    fetch(`${localizedPath(path, config)}.plain.html`).then((res) => {
+      if (res.ok) {
+        res.text().then((cardContent) => {
+          resolve({ path, cardContent: /^<div>(.*)<\/div>$/.exec(cardContent.replaceAll('\n', ''))[1] });
+        });
+      }
+    });
   } catch (error) {
-    err = error.message;
-  }
-  if (!cardsData) {
-    fail(el, err);
+    reject(error);
```

Comment:
if the promo fails, we should still give a change to the default page to render and not reject here.

Reply:
sorry I confused `reject` with `fail` helper.